### PR TITLE
Setup create task

### PR DESCRIPTION
**Pull request by Mr-Smyth**
- Added code to create a user: route template, controller, handler and updated task service
- Updated CSS to display task creation modal correctly
- Updated docStrings

### CODE REVIEW

File: app/components/user-selector.js [hunk 1]
```diff
@@ -7,27 +7,48 @@ export default class UserSelectorComponent extends Component {
...
   }
 
   @action
   handleChange(event) {
-    // Update the selected user ID when the selection changes - store it locally until we grab it in edit-task controller
-    this.selectedUserId = event.target.value;
-
+    // Convert the selected value to null if it represents unassignment, otherwise remain with same value
+    this.selectedUserId =
+      event.target.value === 'null' ? null : event.target.value;
```
Comment (far-fetched) [inline app/components/user-selector.js]: Alternative version might be:
```js
this.selectedUserId = event.target.value ?? null
```
imo slightly easier

File: app/components/user-selector.js [hunk 2]
```diff
@@ -7,27 +7,48 @@ export default class UserSelectorComponent extends Component {
...
-      : [{ id: null, name: 'Unassigned' }, ...this.args.users];
+    const baseOption = {
+      id: null,
+      name: this.selectedUserId ? 'Set as Unassigned' : 'Unassigned',
+    };
+
+    // Map over the real users provided in arguments to format them for the dropdown
+    const userOptions = this.args.users.map((user) => ({
+      id: user.id, // Use user ID as the value
+      name: `${user.firstName} ${user.lastName}`,
```
Comment (far-fetched) [inline app/components/user-selector.js]: if `user` comes as a model, we can use `name` from getter model.

File: app/controllers/users/create-user.js [hunk 3]
```diff
@@ -17,39 +18,34 @@ export default class UsersCreateUserController extends Controller {
...
+    };
+
+    // Save the user to the server via the service
+    try {
+      await this.requestUserService.createUser(newUser);
+      this.closeModal();
+    } catch (error) {
+      console.error('Error creating user:', error);
+      alert('Failed to create user. Please try again later.');
+    }
```
Comment (far-fetched) [inline app/controllers/users/create-user.js]: Definitely better than `then` callback :ok_hand:

**Review by far-fetched — APPROVED** [review]
Tests :heart_eyes:

File: app/components/user-selector.js (hunk 2 above)
Comment (Mr-Smyth) [reply app/components/user-selector.js]: I never actually added a name getter - will add one now to the user model and change that up (what you reference above - thanks for the tipp) 👍🏼

**Pull request merged**